### PR TITLE
Remove web extension library

### LINKED ISSUE #2177: WebExtension support broken?
I tried loading Beastify on master, and if I run

```lisp
(define-configuration web-buffer
  ((default-modes `(beastify ,@%slot-default%))))
```

then the next buffer creation fails with

```log
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: undefined:1:9: ReferenceError: Can't find variable: browser
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: undefined:1:9: ReferenceError: Can't find variable: browser
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: undefined:1:9: ReferenceError: Can't find variable: browser
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: undefined:1:9: ReferenceError: Can't find variable: browser
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: undefined:1:9: ReferenceError: Can't find variable: browser
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: undefined:1:9: ReferenceError: Can't find variable: browser
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: undefined:1:9: ReferenceError: Can't find variable: browser
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: undefined:1:9: ReferenceError: Can't find variable: browser
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: undefined:1:9: ReferenceError: Can't find variable: browser
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: undefined:1:9: ReferenceError: Can't find variable: browser
<INFO> [13:34:52] Buffer 1337 has no history.
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: nyxt:help:1:9: ReferenceError: Can't find variable: browser
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: nyxt:help:1:9: ReferenceError: Can't find variable: browser
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: nyxt:help:1:9: ReferenceError: Can't find variable: browser
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: nyxt:help:1:9: ReferenceError: Can't find variable: browser
<WARN> [13:34:52] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: nyxt:help:1:9: ReferenceError: Can't find variable: browser
```

@aartaka Any idea what's going on?

### LINKED ISSUE #2892: WEBKIT_FORCE_SANDBOX no longer allows disabling the sandbox. Use WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS=1 instead
Starting nyxt git compiled for openSUSE Tumbleweed with WebKitGTK 2.40.0 results in a crash

```
Nyxt version 3
<INFO> [16:50:35] Source location: #P"/usr/share/"
<INFO> [16:50:35] Listening to socket: #P"/run/user/1000/nyxt/nyxt.socket"

** (nyxt:1191): WARNING **: 16:50:35.289: WEBKIT_FORCE_SANDBOX no longer allows disabling the sandbox. Use WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS=1 instead.
<INFO> [16:50:35] GTK extensions directory: #P"/usr/share/libraries/web-extensions/"
<INFO> [16:50:35] GTK extensions directory: #P"/usr/share/libraries/web-extensions/"

** (nyxt:1191): ERROR **: 16:50:35.629: Unable to configure xdg-desktop-portal access in the WebKit sandbox: GApplication is required.
<WARN> [16:50:35] Warning: Error in FFI method: The value
  :INVALID-CODE-OBJECT-AT-PC
is not of type
  (SIMPLE-ARRAY (COMPLEX DOUBLE-FLOAT) (*))

** (process:1191): ERROR (recursed) **: Unable to configure xdg-desktop-portal access in the WebKit sandbox: GApplication is required.
fatal error encountered in SBCL pid 1191 tid 1212:
SIGABRT received.
```

Starting it with `WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS=1` fixes it:

```
WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS=1 nyxt
```

So i suppose code should be updated to use `WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS` for newer WebKitGTK versions.

### PR DESCRIPTION
**Pull request by aartaka**
# Description

This removes the web-extensions library as under-maintained. This also removes all the calls to the Lisp-side of it, and the Lisp-side itself.

Fixes  #2177

# Discussion

@Ambrevar, I've hardcoded `/url/lib/nyxt/` as the library directory in 7bfce963b4cd1522e990f4cb829bd5d04fa64744, which makes all the NASDF library infrastructure unused. Is that the right fix? Should we move it to the NASDF side?

# Checklist:
Everything in this checklist is required for each PR.  Please do not approve a PR that does not have all of these items.

- [X] Git hygiene:
  - I have pulled from master before submitting this PR
  - There are no merge conflicts.
- [X] I've added the new dependencies as:
  - ASDF dependencies,
  - Git submodules,
    ```sh
	cd /path/to/nyxt/checkout
    git submodule add https://gitlab.common-lisp.net/nyxt/py-configparser _build/py-configparser
    ```
  - and Guix dependencies.
- [X] My code follows the style guidelines for Common Lisp code. See:
  - [Norvig & Pitman's Tutorial on Good Lisp Programming Style (PDF)](https://www.cs.umd.edu/~nau/cmsc421/norvig-lisp-style.pdf)
  - [Google Common Lisp Style Guide](https://google.github.io/styleguide/lispguide.xml)
- [X] I have performed a self-review of my own code.
- [x] My code has been reviewed by at least one peer.  (The peer review to approve a PR counts.  The reviewer must download and test the code.)
- [X] Documentation:
  - All my code has docstrings and `:documentation`s written in the aforementioned style.  (It's OK to skip the docstring for really trivial parts.)
  - I have updated the existing documentation to match my changes.
  - I have commented my code in hard-to-understand areas.
  - I have updated the `changelog.lisp` with my changes if it's anything user-facing (new features, important bug fix, compatibility breakage).
  - I have added a `migration.lisp` entry for all compatibility-breaking changes.
  - (If this changes something about the features showcased on Nyxt website) I have these changes described in the new/existing article at Nyxt website or will notify one of maintainters to do so.
- [X] Compilation and tests:
  - My changes generate no new warnings.
  - I have added tests that prove my fix is effective or that my feature works.  (If possible.)
  - I ran the tests locally (`(asdf:test-system :nyxt)` and `(asdf:test-system :nyxt/gi-gtk)`) and they pass.

### CODE REVIEW

File: source/renderer/gtk.lisp [hunk 1]
```diff
@@ -310,14 +309,13 @@ the renderer thread, use `defmethod' instead."
...
-  (:documentation "Directory where to load the 'libnyxt' library.
-By default it is found in the source directory."))
+  (:documentation "Directory where to find WebKitGTK extension .so files."))
 
 (defmethod files:resolve ((profile nyxt-profile) (file gtk-extensions-directory))
   (let ((system-directory (call-next-method)))
     (if (uiop:directory-exists-p system-directory)
         system-directory
-        (files:join (files:expand *source-directory*) "/libraries/web-extensions/"))))
+        "/usr/lib/nyxt/")))
```
Comment (Ambrevar) [inline source/renderer/gtk.lisp:318]: I don't understand, you can leave it to `*source-directory*`, right?  It does not seem to be related to the web-extensions library.

**Comment by Ambrevar** [comment]
As far as _removing_ it goes, the code is fine (beside the `/usr/lib/nyxt/` change maybe).

But do we really want to remove it?  Why not keep it, simply not build it by default?
This way contributors could get inspired by fairly advanced foundations.

**Comment by jmercouris** [comment]
I like the idea of having it, but not building it by default. Just being able to specify it as an option. I do think though that it will continue to drift unless it is in the main build branch...!

**Comment by aartaka** [comment]
Okay, removal of the library reverted, and extension directory is fixed. Merging (I'll obviously squash the commits and reorder them in a logical order before merging)?

**Comment by aadcg** [comment]
Nothing to add from my side. I agree with @jmercouris and @Ambrevar.

**Comment by Ambrevar** [comment]
Did you forget to push?

File: nyxt.asd [hunk 2]
```diff
@@ -426,9 +426,7 @@ The renderer is configured from NYXT_RENDERER or `*nyxt-renderer*'."))
   :pathname #p"NYXT:source;"
   :serial t
   :around-compile "NASDF:FAIL-ON-WARNINGS"
-  :components ((:file "web-extensions")
-               (:file "web-extensions-callbacks")
-               (:file "renderer/gtk-clipboard")
```
Comment (Ambrevar) [inline nyxt.asd:431]: We still need a system to build the web-extensions.

File: source/renderer/gtk.lisp [hunk 3]
```diff
@@ -242,7 +242,6 @@ the renderer thread, use `defmethod' instead."
    on."
   (declare (ignore urls startup-timestamp))
   (log:debug "Initializing GTK Interface")
-  (setf (uiop:getenv "WEBKIT_FORCE_SANDBOX") "0")
```
Comment (Ambrevar) [inline source/renderer/gtk.lisp:245]: Add documentation explaining that this env variable (actually `WEBKIT_FORCE_SANDBOX_THIS_IS_DANGEROUS` in recent versions) may need to be set in order for web extensions to work.

File: source/renderer/gtk.lisp [hunk 4]
```diff
@@ -1475,7 +1463,6 @@ the `active-buffer'."
   (mapc (lambda (handler-id)
           (gobject:g-signal-handler-disconnect (gtk-object buffer) handler-id))
         (handler-ids buffer))
-  (nyxt/web-extensions::tabs-on-removed buffer)
```
Comment (Ambrevar) [inline source/renderer/gtk.lisp:1478]: Same.

File: source/renderer/gtk.lisp [hunk 5]
```diff
@@ -1890,8 +1864,6 @@ local anyways, and it's better to refresh it if a load was queried."
   (not (webkit:webkit-web-view-get-is-muted (gtk-object buffer))))
 #+webkit2-mute
 (defmethod (setf ffi-buffer-sound-enabled-p) (value (buffer gtk-buffer))
-  (nyxt/web-extensions::tabs-on-updated
-   buffer (alex:alist-hash-table `(("audible" . ,value))))
```
Comment (Ambrevar) [inline source/renderer/gtk.lisp:1894]: Same.

**Comment by aartaka** [comment]
Yes, they are. We retain them too? They are drifting into unmaintenance too :(

**Comment by aartaka** [comment]
Okay, all covered and simplified now.

**Comment by Ambrevar** [comment]
I've pushed a note on disabling sandboxing.  Feel free to squash with the previous commit.

I just realized that the `web-extensions` target was never required anywhere in the `makefile`: does that mean that web extension support was not built-in by default anyways?

File: libraries/web-extensions/README.org [hunk 6]
```diff
@@ -9,6 +9,18 @@ via WebKitUserMessages. To hook into the library in a WebKit-based
...
 based on message name -- it will be the same as the JS function name
 (e.g. "runtime.sendMessage").
+* Usage note
+For now WebKit sandboxing must be disabled in order to use the Nyxt
+web-extensions library.  Before starting Nyxt, set the following environment
+variables:
+
+#+begin_src sh
+export WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS=1
+export WEBKIT_DISABLE_SANDBOX=1
```
Comment (aartaka) [inline libraries/web-extensions/README.org:19]: Is there such a variable? Maybe mention `WEBKIT_FORCE_SANDBOX=0` too?

**Comment by aartaka** [comment]
> I've pushed a note on disabling sandboxing. Feel free to squash with the previous commit.

Feels good on its own too.

> I just realized that the `web-extensions` target was never required anywhere in the `makefile`: does that mean that web extension support was not built-in by default anyways?

Hmmmmmmm. Could be, yes.

File: libraries/web-extensions/README.org (hunk 6 above)
Comment (Ambrevar) [reply libraries/web-extensions/README.org:19]: Oops, right.

Comment (aartaka) [reply libraries/web-extensions/README.org:19]: So fix this variable and we can merge, I guess?

**Comment by Ambrevar** [comment]
I've opened #2917 about removing web-extension*.lisp from the default build.
